File: backend/src/main/java/net/ugurkartal/backend/services/abstracts/ProductService.java
package net.ugurkartal.backend.services.abstracts;

import net.ugurkartal.backend.services.dtos.requests.ProductRequest;
import net.ugurkartal.backend.services.dtos.responses.ProductCreatedResponse;
import net.ugurkartal.backend.services.dtos.responses.ProductGetAllResponse;

import java.util.List;

public interface ProductService {
    List<ProductGetAllResponse> getAllProducts();
    List<ProductGetAllResponse> getProductsByCategoryId(String categoryId);
    ProductCreatedResponse getProductById(String id);
    ProductCreatedResponse addProduct(ProductRequest productRequest);
    ProductCreatedResponse updateProduct(String id, ProductRequest productRequest);
    ProductCreatedResponse changeProductStatus(String id, boolean status);
    void updateStock(String id, int quantity);
}

File: frontend/src/services/ProductService.ts
import axios from 'axios';
import {Product} from "../types/Product.ts";

export default class ProductService {
    getAllProducts() {
        return axios.get('/api/products');
    }

    getProductsByCategoryId(categoryId: string) {
        return axios.get(`/api/products/by-category/${categoryId}`);
    }

    getProductById(id: string) {
        return axios.get(`/api/products/${id}`);
    }

    addProduct(product: Product) {
        return axios.post('/api/products', product);
    }

    updateProduct(id:string, product: Product) {
        return axios.put(`/api/products/${id}`, product);
    }

    changeProductStatus(id: string, status: boolean) {
        return axios.put(`/api/products/status/${id}?status=${status}`);
    }
}